src/ng-generate/components/table/generators/directives/resize/files/__name@dasherize__.directive.ts.template

```diff
@@ -69,35 +71,27 @@ export class <%= classify(name) %>Directive implements OnInit {
         this.mouseLeaveListener();
         this.dragging.emit(true);
         this.startX = event.pageX;
-        this.startWidth = this.column.offsetWidth;
+        this.startWidth = this.column.getBoundingClientRect().width;
```

Comment:
Thanks for the catch, `getBoundingClientRect` is indeed less efficient and may lead to a high level of resource consumption, especially when it's getting triggered quite often. I'll check again if the `offsetWidth` solution works for different cases.
I had another one improvement in mind: we can convert it to a stream and use `distinctUntilChanged` in order to reduce a number of re-calculations, but I wasn't able to implement it as a part of the current task because of the time limitations, unfortunately.

Reply:
Comparing to `getBoundingClientRect`, the `offsetWidth` solution provides not so precise measurement, but good enough for the majority of the cases. Based on that, the PR has been updated to use `offsetWidth` approach.